vscode-trace-extension/src/utils/trace-extension-logger.ts

```diff
@@ -0,0 +1,141 @@
+   */
+  disposeChannel(): void {
+      this.logChannel.dispose();
+      this.channelDisposed = true;
+  }
+
+  /**
+   * Adds a log entry to the log channel with an optional tag
+   *
+   * @param message message to be appended to the log channel
+   * @param tag appended to the front of the message (i.e. [$tag] $message)
+   */
+  addLogMessage(message: string, tag?: string): void {
+      let fullMessage: string;
+      if (tag) {
+          fullMessage = '[' + tag + '] ' + message;
+      } else {
+          fullMessage = message;
+      }
+      this.addLogPanelMessage(fullMessage);
+  }
+
+  /**
+   * Adds a log entry for an error.
+   * Prints the error message and the stack
+   *
+   * @param e error to be appended to the log channel
+   */
+  logError(e: Error): void {
+      this.addLogMessage(e.message);
+      if (e.stack) {
+          this.addLogMessage(e.stack);
+      }
+  }
+
+  /**
+   * Shows a information message to user as a notification and
+   * adds the message to log the log channel
+   *
+   * @param message information message to show to the user and add to the log channel
+   * @param options Configures the behavior of the message
+   * @returns void promise
+   */
+  async showInfo(message: string, options?: vscode.MessageOptions): Promise<void> {
+      const fullMessage = 'Info: ' + message;
+      vscode.window.showInformationMessage(fullMessage, options || {modal: false});
+      return this.addLogMessage(fullMessage);
+  }
+
+  /**
+   * Shows a warning message to user as a notification and
+   * adds the message to log the log channel
+   *
+   * @param message warning message to show to the user and add to the log channel
+   * @param options Configures the behavior of the message
+   * @returns void promise
+   */
+  async showWarning(message: string, options?: vscode.MessageOptions): Promise<void> {
+      const fullMessage = 'Warning: ' + message;
+      vscode.window.showWarningMessage(fullMessage, options || {modal: false});
+      return this.addLogMessage(fullMessage);
+  }
+
+  /**
+   * Shows an error message to user as a notification and
+   * adds the message to log the log channel
+   *
+   * @param message error message to show to the user and add to the log channel
+   * @param options Configures the behavior of the message
+   * @returns void promise
+   */
+  async showError(message: string, options?: vscode.MessageOptions): Promise<void> {
+      const fullMessage = 'Error: ' + message;
+      vscode.window.showErrorMessage(fullMessage, options || {modal: false});
+      return this.addLogMessage(fullMessage);
+  }
+
+  /**
+   * Appends the message with a timestamp to the log channel
+   * @param message message to be added to the log channel
+   */
+  private addLogPanelMessage(message: string) {
+      if (!this.channelDisposed) {
+          this.logChannel.append(`[${new Date().toLocaleTimeString('en-US', {hour12: false})}] ${message}\n`);
```

Comment:
We should log also the time. I see other plug-ins doing the same. I will be useful if we log more things and want to trouble-shoot.

Reply:
It does have time.. did you mean date?

![ss1](https://user-images.githubusercontent.com/113199503/233463739-1d24fa90-7a23-411e-83c7-b1d9cc6f0694.png)

Reply:
Oh, true, the date is missing not the time. Right now we don't need more details. But you could change it right away for the future.  I checked a few outputs from other extension and most show something like this: 2023-04-20 13:37:24.500.

Reply:
I have modified the timestamp to be in the same format as the one you mentioned